fennel/datasets/datasets.py

```diff
@@ -723,6 +729,50 @@ def dsschema(self):
         return input_schema
 
 
+class Window(_Node):
+    def __init__(
+        self,
+        node: _Node,
+        keys: List[str],
+        type: str,
+        gap: Duration,
+        field: str,
+    ):
+        super().__init__()
+        if len(keys) == 0:
+            raise ValueError("Must specify at least one key")
+        keys.append(field)
+        self.keys = keys
+        self.type = type
+        self.gap = gap
+        self.gap_timedelta = duration_to_timedelta(gap)
+        self.field = field
+        self.node = node
+        self.node.out_edges.append(self)
+
+    def signature(self):
+        return fhash(
+            self.node.signature(),
+            self.keys,
+            self.type,
+            self.gap,
+            self.field,
+        )
```

Comment:
Do we still use signature/fhash etc anywhere on the backend? If not, time to deprecate?

Reply:
+1 was telling @nonibansal the same. We do populate it and send it to the backend, but its pointless and we should get rid of it.

Reply:
We can get rid on them in another PR, I will create a task for tracking this.
https://linear.app/fennel-ai/issue/DEV-2715/get-rid-of-signaturefhash-in-python-client